src/containers/ViewerPanel/index.tsx

```diff
@@ -232,32 +238,58 @@ class ViewerPanel extends React.Component<ViewerPanelProps, ViewerPanelState> {
 
     public onTrajectoryFileInfoChanged(data: TrajectoryFileInfo) {
         const { receiveMetadata, simulariumController } = this.props;
-        const { spatialUnitFactorMeters } = data;
 
         const tickIntervalLength = simulariumController.tickIntervalLength;
-        // Format scale bar length and unit so that it's more readable, e.g.:
-        // 0.000000015 m -> [15, "nm"]
-        const scaleBarLabelArray = si.meter.convert(
-            tickIntervalLength * spatialUnitFactorMeters
-        );
-        const scaleBarLabelNumber: number = parseFloat(
-            scaleBarLabelArray[0].toPrecision(2)
-        );
-        // The si-prefix library abbreviates "micro" as "mc", so swap it out with "µ"
-        const scaleBarLabelUnit: string = scaleBarLabelArray[1].replace(
-            "mc",
-            "µ"
-        );
+        let scaleBarLabelNumber: number;
+        let scaleBarLabelUnit: string;
+
+        switch (data.version) {
+            case 1:
+                const dataV1 = data as TrajectoryFileInfoV1;
+
+                // Format scale bar length and unit so that it's more readable, e.g.:
+                // 0.000000015 m -> [15, "nm"]
+                const scaleBarLabelArray = si.meter.convert(
+                    tickIntervalLength * dataV1.spatialUnitFactorMeters
+                );
+                scaleBarLabelNumber = parseFloat(
+                    scaleBarLabelArray[0].toPrecision(2)
+                );
+                // The si-prefix library abbreviates "micro" as "mc", so swap it out with "µ"
+                scaleBarLabelUnit = scaleBarLabelArray[1].replace("mc", "µ");
+
+                receiveMetadata({
+                    numFrames: dataV1.totalSteps,
+                    timeStepSize: dataV1.timeStepSize,
+                    timeUnits: null,
+                });
+
+                break;
+
+            case 2:
+                const dataV2 = data as TrajectoryFileInfoV2;
+
+                scaleBarLabelNumber =
+                    tickIntervalLength * dataV2.spatialUnits.magnitude;
+                scaleBarLabelUnit = dataV2.spatialUnits.name;
+
+                receiveMetadata({
+                    numFrames: dataV2.totalSteps,
+                    timeStepSize:
+                        dataV2.timeStepSize * dataV2.timeUnits.magnitude,
+                    timeUnits: dataV2.timeUnits,
+                });
```

Comment:
This again seems like it shouldn't be in a react component, even if it is a container. I think it could be a good idea to have a version manager utility folder (probably in util/) that has functions for conditionally formatting the data based on version number. My worry about having this here is that when a version changes again or we want to remove them, we'll have to be looking around for all the files that mention versions. In this case it would be a `formatTrajectoryTimeInfoByVersion` or something that takes the `data` here returns the `numFrames` `timeStepSize` and `timeUnits`

Reply:
That sounds good. I'm thinking I'll add a new file called `versionHandlers.ts` (or something like that) under `util/` and add 2 functions: 
* one that takes in `tickIntervalLength`, `data`, and `version` and returns a `scaleBarLabel`
* one that takes in `data` and `version` and returns `numFrames`, `timeStepSize`, and `timeUnits` like you said.

Reply:
I can also save the version number as metadata to redux here so that I can use that version number in a redux selector to do the things I'm currently doing in `PlaybackControls`, does that sound good? 

And if I hook up PlaybackControls to redux with selectors I guess I'll need to make PlaybackControls a container. Wonder if we can consider doing away with containers/components distinction like Gabe did for the FMS projects a few weeks ago, not on this ticket though.

Reply:
Better idea than what I just said: I'll try to leave PlaybackControl a component (not connected to redux) and handle everything in ViewerPanel/redux, and pass down `roundedTime`, `roundedLastFrameTime`, and `unitLabel` to PlaybackControl as regular props

Reply:
yeah, I definitely dont think this component needs to be connected to redux, what you said sounds good